client/src/test/suite/extension.test.ts

```diff
@@ -1008,4 +1010,239 @@ suite('Push Button Test Suite', function (this: Suite) {
     pushStub.reset()
     showInfoMsgStub.reset()
   })
+  test('validateContent only runs when it should', async () => {
+    const showInformationMessageStub = sinon.stub(vscode.window, 'showInformationMessage')
+    const openAndValidateStub = sinon.stub(pushContent, 'openAndValidate')
+
+    showInformationMessageStub.resolves(undefined)
+    await pushContent.validateContent()
+    assert(openAndValidateStub.notCalled)
+    openAndValidateStub.reset()
+
+    // 'as any' is required here because of showQuickPick overloading
+    showInformationMessageStub.resolves(pushContent.DocumentsToOpen.all as any)
+    await pushContent.validateContent()
+    assert(openAndValidateStub.calledWith(pushContent.DocumentsToOpen.all))
+  })
+  test('getDocumentsToOpen returns all files', async () => {
```

Comment:
Could some of these tests be written as Jest unit tests? These new tests are already stubbing several methods. Also, since we switched to Jest I've been chiseling away at this still-gigantic file